cmd/cstor-volume-mgmt/volume/volume.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/util"
+)
+
+// VolumeOperator is the name of the tool that makes volume-related operations.
+const (
+	VolumeOperator = "iscsi"
+	IstgtConfPath  = "/usr/local/etc/istgt/istgt.conf"
+)
+
+//RunnerVar is
+var RunnerVar util.Runner
+
+//FileOperatorVar is
+var FileOperatorVar util.FileOperator
+
+//UnixSockVar is
+var UnixSockVar util.UnixSock
+
+// CreateVolume creates a new cStor volume istgt config.
+func CreateVolume(cStorVolume *apis.CStorVolume) error {
+
+	//generate sparse file
+	// generateSparseFile(cStorVolume)
+	touchArgs := sparseFileCommandBuilder(cStorVolume, "touch")
+
+	stdoutStderr, err := RunnerVar.RunCombinedOutput("/usr/bin/touch", touchArgs...)
+	if err != nil {
+		glog.Error("failed to touch file /tmp/cstor/"+cStorVolume.Spec.VolumeName,
+			err.Error(), string(stdoutStderr))
+		return err
+	}
+
+	truncArgs := sparseFileCommandBuilder(cStorVolume, "truncate")
+
+	stdoutStderr, err = RunnerVar.RunCombinedOutput("/usr/bin/truncate", truncArgs...)
+	if err != nil {
+		glog.Error("failed to truncate file /tmp/cstor/"+cStorVolume.Spec.VolumeName+
+			" with capacity "+cStorVolume.Spec.Capacity,
+			err.Error(), string(stdoutStderr))
+		return err
+	}
+
+	// create conf file
+	text := CreateIstgtConf(cStorVolume)
+	err = FileOperatorVar.Write(IstgtConfPath, text, 0644)
+	if err != nil {
+		glog.Errorf("Failed to write istgt.conf...")
+	}
+	glog.Info("Done writing istgt.conf")
+
+	// send refresh command to istgt
+	err = UnixSockVar.SendCommand("REFRESH\n")
+	if err != nil {
+		glog.Info("refresh failed")
+	}
+	glog.Info("Creating Iscsi Volume Successful")
+	return nil
+
+}
+
+// sparseFileArgumentsBuilder is to build sparse file command.
+func sparseFileCommandBuilder(cStorVolume *apis.CStorVolume, op string) []string {
+	var cmdArgs []string
+	if op == "touch" {
+		cmdArgs = append(cmdArgs, "/tmp/cstor/"+cStorVolume.Spec.VolumeName)
+	} else if op == "truncate" {
+		cmdArgs = append(cmdArgs, "-s")
+		cmdArgs = append(cmdArgs, cStorVolume.Spec.Capacity)
+		cmdArgs = append(cmdArgs, "/tmp/cstor/"+cStorVolume.Spec.VolumeName)
+	}
+
+	return cmdArgs
+}
+
+// CreateIstgtConf creates istgt.conf file
+func CreateIstgtConf(cStorVolume *apis.CStorVolume) []byte {
+	text := []byte(`
```

Comment:
Does it make sense to use go templating? I strongly believe we should make use of templating. This will make things more clear & understood.
Do you feel, this can form a part of CAS template? CAS template will enable operators to finetune/adjust this conf based on workloads etc.

Reply:
In addition, Does it make sense to use a ConfigMap to store this file?

Reply:
Can we take this up as a separate PR/task ?